.github/workflows/checks.yaml

```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
-          toolchain: stable
+          toolchain: nightly-2025-03-19
```

Comment:
Nit: remove `toolchain` param entirely and then the action should just pull it from `rust-toolchain.toml` file. This would require less things to keep track of next time we upgrade nightly.